src/main/resources/db/migration/V0002__purge.sql

```diff
@@ -0,0 +1,209 @@
+(purge_job_id SERIAL PRIMARY KEY,
+job_start_date TIMESTAMP(6),
+job_end_date TIMESTAMP(6),
+retention_date TIMESTAMP(6),
+count_of_error_logs NUMERIC,
+count_of_individual_requests NUMERIC,
+count_of_bulk_submissions NUMERIC,
+count_of_service_requests NUMERIC,
+success NUMERIC);
+
+CREATE TABLE sdt_owner.purge_job_audit_messages
+(purge_job_message_id SERIAL PRIMARY KEY,
+purge_job_id NUMERIC,
+created_date TIMESTAMP(6),
+log_message VARCHAR(256));
+
+CREATE OR REPLACE FUNCTION sdt_owner.purge (i_CommitInterval IN INT) RETURNS VOID language plpgsql AS $$
+DECLARE
+DECLARE i_retention_period VARCHAR(32);
+DECLARE t_retention_date TIMESTAMP;
+DECLARE t_job_start_date TIMESTAMP;
+DECLARE t_job_end_date TIMESTAMP;
+DECLARE i_purge_job_id NUMERIC;
+
+  -- cursor1 service requests created before given retention date
+  c1 CURSOR (n_param DOUBLE PRECISION)
+            FOR SELECT service_request_id
+               FROM sdt_owner.service_requests
+               WHERE request_timestamp < t_retention_date;
+
+  -- cursor2 bulk submissions with service requests created before given retention date
+  c2 CURSOR (n_param DOUBLE PRECISION)
+            FOR SELECT bulk_submission_id
+               FROM sdt_owner.bulk_submissions
+               WHERE service_request_id IN (
+                   SELECT  service_request_id
+                   FROM sdt_owner.service_requests
+                   WHERE request_timestamp < t_retention_date);
+
+  -- cursor3 individual requests for bulk submissions
+  --   with service requests created before given retention date
+  c3 CURSOR (n_param DOUBLE PRECISION)
+            FOR SELECT individual_request_id,sdt_request_reference
+               FROM sdt_owner.individual_requests
+               WHERE bulk_submission_id IN  (
+                   SELECT bulk_submission_id
+                   FROM sdt_owner.bulk_submissions
+                   WHERE service_request_id IN (
+                       SELECT service_request_id
+                       FROM sdt_owner.service_requests
+                       WHERE request_timestamp < t_retention_date));
+
+  -- cursor4 error logs for individual requests for bulk submissions
+  --   with service requests created before given retention date
+  c4 CURSOR (n_param DOUBLE PRECISION)
+            FOR SELECT error_log_id
+               FROM sdt_owner.error_logs
+               WHERE individual_request_id IN (
+                   SELECT individual_request_id
+                   FROM sdt_owner.individual_requests
+                   WHERE bulk_submission_id IN (
+                       SELECT bulk_submission_id
+                       FROM sdt_owner.bulk_submissions
+                       WHERE service_request_id IN (
+                           SELECT service_request_id
+                           FROM sdt_owner.service_requests
+                           WHERE request_timestamp < t_retention_date)));
+
+  n_iteration   INT;
+  BEGIN
+
+    -- retention period
+	SELECT parameter_value
```

Comment:
Indentation is a bit inconsistent here and on subsequent lines (e.g. indentation used in 'for' loops is 2 spaces, insert statements have same indentation for all lines whereas update statements have extra indentation on second and subsequent lines).  Could it be tidied up?